### PR TITLE
Backport of 6553/6875 to main-7.0.x

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #11639  

Link to tickets:
-  https://redmine.openinfosecfoundation.org/issues/6555
- https://redmine.openinfosecfoundation.org/issues/7224

Describe changes:
- Cherry-pick commits from https://github.com/OISF/suricata/pull/10654 for issue 6555
- Cherry-pick fix for 6875 to avoid dbv error.
- s-v updates for 7.0.7+

Updates:
- Added comments to the commit "membuffer: use buffer pointer as flexible array member" stating why the buffer needs to be reset when created.

### Provide values to any of the below to override the defaults.

- To use an LibHTP, Suricata-Verify or Suricata-Update pull request,
  link to the pull request in the respective `_BRANCH` variable.
- Leave unused overrides blank or remove.

SV_REPO=
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2012
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22370

File: src/util-buffer.c
Comment (victorjulien) [inline src/util-buffer.c:51]: a simpler fix would have been to use SCCalloc above?

Comment (jlucovsky) [reply src/util-buffer.c:51]: I think it'd be equivalent. I like the explicit reset (and it's an inline function).

Your call though -- which would you prefer?

**Review by victorjulien — APPROVED** [review]

File: src/util-buffer.c
Comment (victorjulien) [reply src/util-buffer.c:51]: Prefer less code, so SCCalloc.

Would consider it a nit, but now that we need another round because of the SV test issue, I prefer this to be addressed as well.

**Comment by victorjulien** [comment]
This PR is blocked by the SV PR being in "changes requested" state.

**Review by victorjulien — CHANGES_REQUESTED** [review]
just a nit inline, but needs a SV test update

**Comment by jlucovsky** [comment]
Continued in #11761

**Pull request closed without merging**
